sharding/shard.go

```diff
@@ -68,9 +69,14 @@ func (s *Shard) CollationByHash(headerHash *common.Hash) (*Collation, error) {
 	if err != nil {
 		return nil, fmt.Errorf("cannot fetch body by chunk root: %v", err)
 	}
-	// TODO: deserializes the body into a txs slice instead of using
+
+	// deserialize the body into a txs slice instead of using
 	// nil as the third arg to MakeCollation.
-	col := NewCollation(header, body, nil)
+	txs, err := DeserializeBlobToTx(body)
```

Comment:
Why does this have to do with construct collation and merklize chunks into chunk root?